spec/active_record/key_spec.rb

```diff
@@ -44,7 +44,7 @@ module ActiveRecord
       end
       context 'different ways of setting keys' do
         let(:attrs) { {make: 'Ford', name: 'Focus', long_field: SecureRandom.uuid} }
-        before { Vehicle.create(attrs) }
+        let!(:vehicule) { Vehicle.create(attrs) }
```

Comment:
Would it work with `before`?

Reply:
The removed line `before { Vehicle.create(attrs) }` was useless, it should have been written `before { @vehicle = Vehicle.create(attrs) }`.

Changing `let!(:vehicule) { Vehicle.create(attrs) }` by `let(:vehicule) { Vehicle.create(attrs) }` recreate `vehicle` object on each tests. May be it's better for tests integrity...

Reply:
The current documentation on `let` and `let!`:

https://relishapp.com/rspec/rspec-core/v/3-7/docs/helper-methods/let-and-let

Let's keep the French wording in there. And use `let!`. In this case, we would like to recalculate before each test, so `let!` it is.

Reply:
@olleolleolle you're right